tests/rados/test_scrub_parameters.py

```diff
@@ -0,0 +1,252 @@
+        log.info(e)
+        log.info(traceback.format_exc())
+        return 1
+    finally:
+        log.info("Execution of finally block")
+        method_should_succeed(rados_object.delete_pool, pool_name)
+        rados_object.log_cluster_health()
+    return 0
+
+
+def create_pool_get_osd(rados_object, pool_type, pool_config):
+    """
+    Method is used -
+        1. Create replicated pool
+        2. Push the data into pool
+        3. From the active OSD list return a random OSD
+    Args:
+        rados_object: Rados object
+        pool_name: pool name
+    Returns:
+        Return a random OSD ID from an active OSD list
+    """
+    pool_name = pool_config["pool_name"]
+    if pool_type == "replicated_pool":
+        if not rados_object.create_pool(**pool_config):
+            log.error("Failed to create the replicated Pool")
+            return 1
+    else:
+        if not rados_object.create_erasure_pool(name=pool_name, **pool_config):
+            log.error("Failed to create the EC Pool")
+            return 1
+
+    if not rados_object.bench_write(
+        pool_name=pool_name, byte_size="5M", max_objs=100, verify_stats=True
+    ):
+        log.error("Failed to write objects into Pool-1, with compression enabled")
+        return 1
+    res = wait_for_clean_pg_sets(rados_object, test_pool=pool_name)
+    if not res:
+        log.error("PG's in cluster are not active + Clean ")
+        return 1
+    osd_list = rados_object.get_active_osd_list()
+    osd_id = random.choice(osd_list)
+    return osd_id
+
+
+def chk_forced_flag(scrub_object, osd_id):
+    """
+    Method is used to check the forced flag status
+    Args:
+        rados_object: Rados object
+        osd_id: OSD ID number
+    Returns:
+        True -> If forced flag is true
+        False -> If forced flag is false
+    """
+    pg_id = ""
+    time_execution = datetime.datetime.now() + datetime.timedelta(minutes=20)
+    found = False
+    while datetime.datetime.now() < time_execution:
+        out_put = scrub_object.get_dump_scrubs(osd_id)
+        for pg_no in out_put:
+            if pg_no["forced"] is True:
+                pg_id = pg_no["pgid"]
+                log.info(f"The forced flag is true for the pg id-{pg_id}")
+                found = True
+                return True, pg_id
+        log.info("Scrub/Deep-scrub verification is in-progress")
+        time.sleep(30)
+    if not found:
+        return False, pg_id
+
+
+def wait_pg_forced_flag(scrub_object, osd_id, pg_id):
+    """
+    Method is used to wait for the forced flag status
+    Args:
+        scrub_object: Scrub object
+        osd_id: OSD ID number
+        pg_id: pg id number
+    Returns:
+        True -> If forced flag set to  false
+        False -> If forced flag is true
+    """
+    time_execution = datetime.datetime.now() + datetime.timedelta(minutes=60)
+    while datetime.datetime.now() < time_execution:
+        pg_dump_scrub = scrub_object.get_pg_dump_scrub(osd_id, pg_id)
```

Comment:
Need to check if `get_pg_dump_scrub()` returned `None`. If we proceed further without checking the return value, we might end wu with a exception here.

Reply:
I added code to check the dump scrub output.